src/bentoml/_internal/io_descriptors/utils.py

```diff
@@ -0,0 +1,25 @@
+from typing import AsyncGenerator
+from typing import AsyncIterator
+
+
+def to_sse(
+    data: str, event: str | None = None, id: str | None = None, retry: str | None = None
+) -> str:
+    sse_data = ""
+    if id is not None:
+        sse_data += f"id: {id}\n"
+    if event is not None:
+        sse_data += f"event: {event}\n"
+    if retry is not None:
+        sse_data += f"retry: {retry}\n"
+    sse_data += f"data: {data}\n\n"
+    return sse_data
+
+
+async def from_sse(async_iterator: AsyncIterator[str]) -> AsyncGenerator[str, None]:
```

Comment:
This is not really from sse right? rather from iterator that yields sse format.

We can do something like this 

https://github.com/bentoml/OpenLLM/blob/main/openllm-client/src/openllm_client/_stream.py

Reply:
Yeah the naming of the function is probably inaccurate. It's just a wrapper iterator that will combine chunks(if they are chunked) into a complete SSE event. so that we can just do `for event in from_sse(runner.async_stream.xx()`

Wasn't exactly thinking about implementing a new data structure here but mere simple functions. 
since OpenLLM would be mainly using these utils, maybe you could state your preferences

Reply:
I wonder when users will actually using this

```python
bentoml.io.from_see(runner.generate.async_stream())
```
?

This means `async_stream` should either yield a sse object, or a stream with sse format

Preferably to use the object.

Reply:
The decision from @ssheng and @bojiang discussion the other day was to keep the `async_stream` as pure text stream and if we expect an SSE format text stream, then we should wrap it with this `from_sse` method, or whatever its called, otherwise we don't have guarantee that the event will not be chunked. 

CMIIW

Reply:
If async_stream yield pure text_stream then `from_sse` expects the generator is a text-stream with SSE format and then should yield a SSE object (similar to what I did with `openllm-client`)

Reply:
I don't really see the use of these utilities to be frank.

Reply:
okay, if you as the user wouldn't use it we can exclude it from this PR. User will then have the responsibility to patch the text chunks if they are being chunked